pkgs/by-name/hy/hypridle/package.nix

```
@@ -0,0 +1,44 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, pkg-config
+, cmake
+, wayland
+, wayland-protocols
+, hyprlang
+, sdbus-cpp
+, systemd
+}:
+stdenv.mkDerivation (finalAttrs: {
+  pname = "hypridle";
+  version = "0.1.0";
+
+  src = fetchFromGitHub {
+    owner = "hyprwm";
+    repo = "hypridle";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-0x5R6v82nKBualYf+TxAduMsvG80EZAl7gofTIYtpf4=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+  ];
+
+  buildInputs = [
+    hyprlang
+    sdbus-cpp
+    systemd
+    wayland
+    wayland-protocols
+  ];
+
+  meta = {
+    description = "Hyprland's idle daemon";
+    homepage = "https://github.com/hyprwm/hypridle";
+    license = lib.licenses.bsd3;
+    maintainers = with lib.maintainers; [ iogamaster ];
+    mainProgram = "hypridle";
+    inherit (wayland.meta) platforms;
```

```suggestion
    platforms = [ "aarch64-linux" "x86_64-linux" ];
```
Upstream appears to only support the platforms above.

---

Not necessarily, these are just what we have in the flake, as no one else tested/felt the need to add other platforms.

---

If other platforms aren't tested on, how would they be supported?

---

Since hypridle follows the wayland spec, I don't see why it wouldn't work on any platforms wayland does.

---

It's better to have it like this. Literally 5 mins ago some issues were closed about hyprland, xdph and hyprlock not building on 32-bit architectures.